addons/base_vat/models/res_partner.py

```diff
@@ -598,6 +599,15 @@ def check_vat_au(self, vat):
             return len(vat) == 11 and vat.isdigit()
         return check_func(vat)
 
+    def check_vat_nz(self, vat):
+        '''
+        The New Zealand equivalent of a VAT number is an IRD number (GST number is another name for this).
+        IRD/GST numbers must legally must be displayed on all tax invoices.
+        https://arthurdejong.org/python-stdnum/doc/1.13/stdnum.nz.ird#module-stdnum.nz.ird
+        '''
+        check_func = getattr(stdnum.util.get_cc_module('nz', 'ird'), 'is_valid', None)
```

Comment:
@rjocoleman Indeed, I wonder why stdnum does not return something in 
 `getattr(stdnum.util.get_cc_module('nz', 'vat'), 'is_valid', None)`

Reply:
The stdnum alias was added in 1.14 - https://github.com/arthurdejong/python-stdnum/blob/master/stdnum/nz/__init__.py#L24

and Odoo's requirements on this branch lists python-stdnum==1.13
 
Stdnum's current version is 1.18